includes/Core/Util/Authentication_Trait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Trait Google\Site_Kit\Core\Util\Authentication_Trait
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+/**
+ * Trait for getting fake authentication credentials.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+trait Authentication_Trait {
+
+	/**
+	 * Creates fake Google authentication credentials for testing.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function fake_authentication() {
+		add_filter(
+			'googlesitekit_oauth_secret',
+			function () {
+				return wp_json_encode(
```

Comment:
This is somewhat unrelated but would be good to update here - we no longer require that the secret is JSON encoded - it can be pre-decoded (as an array) so we can simply remove the wrapping `wp_json_encode` call here and return the array as-is. Same below 👍

Reply:
Fixed